### ISSUE TITLE
1.0.0 required for SDK?

### ISSUE THREAD

**Issue by mixmix** [hyperledger-identus/mediator#312]
We're trying to develop with `@atala/prism-wallet-sdk` v5 and the [releases repo](https://github.com/input-output-hk/atala-releases/blob/master/Atala%20PRISM/2.12.md) suggests to use `MEDIATOR_VERSION=0.14.2` .... but using that version of the repo leads to errors in the SDK - it won't start up

```bash
 Error: Invalid state: Unable resolve sender secret: Unable get secret: Invalid PrivateKey Curve X25519
  at BasicMediatorHandler.achieveMediation (/home/NAME/ssb-atala-prism/node_modules/@atala/prism-wallet-sdk/build/node/index.cjs:11651:27)
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  at async ConnectionsManager.registerMediator (/home/NAME/ssb-atala-prism/node_modules/@atala/prism-wallet-sdk/build/node/index.cjs:11019:9)
  at async Agent.start (/home/NAME/ssb-atala-prism/node_modules/@atala/prism-wallet-sdk/build/node/index.cjs:11879:17)
```

We tried a lot of different things, including using the `MEDIATOR_DID` referenced in the SDK demos ([here](https://github.com/input-output-hk/atala-prism-wallet-sdk-ts/blob/main/demos/node-esm/src/index.js#L9)). Interestingly, that one works fine and the SDK agent is able to start up! Looking closer at that mediator we see the DID resolves to https://sit-prism-mediator.atalaprism.io/ ... and scrolling down we see that it's running version `1.0.0` :skull:

**Comment by mixmix** [comment]
SO... something is out of step:
1. either our Mediator `0.14.2` is mis-configured (impossible for me to say from that error message)
2. your releases repo is wrong?
3. your SDK is being tested against the wrong mediator?

Looking at https://github.com/input-output-hk/atala-prism-wallet-sdk-ts/blob/v5.1.1/.github/workflows/e2e-tests.yml#L15 we can see that yes your CI is running against this version `1.0.0` mediator.

I still don't know if this is the source of the Error. But the confusion in this neighborhood is making our lives really hard

**Comment by mixmix** [comment]
Finally, looking at the `identus-mediator-v1.0.0` tag docker compose https://github.com/input-output-hk/atala-prism-mediator/blob/identus-mediator-v1.0.0/docker-compose.yml we see that the docker image has moved : 

```diff
-  atalaprism-mediator:
-    image: ghcr.io/input-output-hk/atala-prism-mediator:${MEDIATOR_VERSION:-latest}
+  identus-mediator:
+    image: ghcr.io/input-output-hk/identus-mediator:${MEDIATOR_VERSION:-latest}
```

AND.... it's not accessible (to me). Here's the error when we try to pull it:
```bash
 ✘ identus-mediator Error Head "https://ghcr.io/v2/input-output-hk/identus-mediator/manifests/1.0.0": un...                         2.1s
Error response from daemon: Head "https://ghcr.io/v2/input-output-hk/identus-mediator/manifests/1.0.0": unauthorized
```

**Comment by mixmix** [comment]
```
image: ghcr.io/input-output-hk/atala-prism-mediator:${MEDIATOR_VERSION:-latest}
```
with
```
MEDIATOR_VERSION=1.0.0
```

works for pulling an image ... but when trying to start the SDK we get:

```bash

Error: Malformed: Message is not a valid JWE, JWS or JWM
  at BasicMediatorHandler.achieveMediation (/home/projects/ssb-atala-prism/node_modules/@atala/prism-wallet-sdk/build/node/index.cjs:11646:27)
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  at async ConnectionsManager.registerMediator (/home/projects/ssb-atala-prism/node_modules/@atala/prism-wallet-sdk/build/node/index.cjs:11019:9)
  at async Agent.start (/home/projects/ssb-atala-prism/node_modules/@atala/prism-wallet-sdk/build/node/index.cjs:11874:17)
```
Once again an Error that tells me nothing about what's gone wrong or how to fix it

**Comment by mineme0110** [comment]
@mixmix 

Version 1.0.0 is not the correct version released. We are currently in the process of migrating the libraries to identus-mediator. The stable version is prism-mediator-v0.14.2, and this should be used until the repositories are migrated. All other versions can be ignored for now.
FYI prism-mediator will be migrated  as identus-mediator  under this  https://github.com/hyperledger/

**Comment by FabioPinheiro** [comment]
About `Malformed: Message is not a valid JWE, JWS or JWM` that is very familiar to me.
I reported and help fix two error on other libs that produced this error.
@mixmix What DID Comm lib and version are you using in your application?

**Comment by mixmix** [comment]
@FabioPinheiro that's a 1.0.0 mediator version trying to talk with sdk 5.1.1


@mineme0110
1. what is the error:  `Error: Invalid state: Unable resolve sender secret: Unable get secret: Invalid PrivateKey Curve X25519`
2. Why is the sdk CI not testing against 0.14.2 mediator (does it even pass tests against that?)

**Comment by FabioPinheiro** [comment]
@mixmix `Malformed: Message is not a valid JWE, JWS or JWM ` i believe the error is from your DID Comm lib that you are using in the javascript. I was asking what is the version and library that you are using on your application.

The version 1.0.0 mediator should not exist, at least for now. It was created because people like automations that does more than it should. I cannot even delete the tag because it would be created again by the CI;CD ...
Currently we are blocked until the repository is moved so we can do things properly

1) can you provide more context, where you are from?

2) Yes 0.14.2 pass all the tests and integration tests

**Comment by chereseeriepa** [comment]
Hi @FabioPinheiro im working alongside @mixmix on this.

Context: We are from Ahau/Matou and have been writing end-to-end integration tests [here](https://gitlab.com/matou-collective/lib/atala-prism/ssb-atala-prism/-/merge_requests/1)

We are using [atala-prism-wallet-sdk-ts](https://github.com/input-output-hk/atala-prism-wallet-sdk-ts/blob/v5.1.1/package.json#L98-L99) `v5.1.1` which comes bundled with its own DID Comm libraries which we assume work.

**Comment by FabioPinheiro** [comment]
In that case it should work. I'm not aware of any incompatible
I have see the error `Malformed: Message is not a valid JWE, JWS or JWM` several times for different reasons. But they all boil down to the fact that the library is not capable of being parse the field of the DID Comm message (missing field or wrong structure).

I would be happy to debug this if I had the steps to reproduce.
I can even that I very quick look with my tools https://did.fmgp.app/. To see if we could spot the problem before starting to debug. I just need the encrypted message itself and the private keys.

**Comment by elribonazo** [comment]
If i can propose would love to work on this in a single thread or place. This is the mediator while the issue is more on the Typescript SDK, I'm thinking in the storage probably but need some more work.

So there is really couple things here, let me try simplifying and grouping the issues.

1. Mediator & repository migration to Hyperledger: well known already that we are going to migrate the repositories to Hyperledger.
As @FabioPinheiro is suggesting, stick to using 0.14.2, without caring too much on HyperLedger or not, both should have that version.

      Small suggestion: _I'm checking the test files and configuration u guys are building and all the SH scripts, those are likely to change over time and don't think its a good idea. Using submodules would probably be better.
      The environment files are going to be kept in your control its just that the standard configuration and running scripts will run for you from the sub-module. Also u'd ofc keep control over the exact commit that u want to be using._
      
      
2. Malformed: Message is not a valid JWE, JWS or JWM
This is something we've had ongoing for a while and in the most recent version (not yet released) I'm mitigating the error which most of the times can just be ignored but is bothering and affecting some flows like e2e tests if that isnn't kept in mind.

    Has been fixed recently [here](https://github.com/input-output-hk/atala-prism-wallet-sdk-ts/blob/e80549cb08082ccc72b0a4abc6ba2e539d9e6fb7/src/mercury/Mercury.ts#L130) and going to ask the team to release it quickly.
    
    Most of the times it happens when reaching out to the didcomm endpoint in the enterprise agent so its really not the mediator endpoint. Mediator also could potentially return something that is not a valid Message, on any case the issue is now mitigated and this exception will not throw, if the message cannot be parsed instead will return null avoiding the exception
    
3. Error: Invalid state: Unable resolve sender secret: Unable get secret: Invalid PrivateKey Curve X25519
This is storage related for sure, need to see your case and debug what is it what you guys are getting from store. Looks to me like the Key cannot be found, the sender key. But as this is really related to the way the lib is using we'll need to jump in shortly to your repo and investigate there.

**Comment by elribonazo** [comment]
@mixmix I've jumped into your repository and made couple changes in a new branch called docker-fixes.

1. Upgraded the SDK to the latest version which is just a new feature no breaking changes.
2. Added the proper configuration for local mediator, does not matter 1.0.0 or 0.14.2 both work and achieve mediation without issues.

Replicating the configuration as i said earlier has some risks :) and in this case, the risk was not copying the initdb.js, but instead creating a folder with the same name. MongoDB starts with a failure and the mediator cannot connect to the database, that is why the mediator was returning null and the SDK cannot really continue in this case. 

If the OAS returns "" or "null" now it can be ignored anyways.
Still see a bunch of tests failing but they don't seem to be related to errors or misconfiguration, probably u guys can check those.

let's organize a call together next week1 and I'd also propose to close the issue.
